src/context/URQL.tsx

```diff
@@ -2,24 +2,31 @@ import type { BaseComponentProps } from 'src/types';
 
 import { useMemo } from 'react';
 
+import { useShallow } from 'zustand/react/shallow';
+
 import { authExchange } from '@urql/exchange-auth';
 import { cacheExchange } from '@urql/exchange-graphcache';
 import { requestPolicyExchange } from '@urql/exchange-request-policy';
 import { DateTime } from 'luxon';
 import { Client, fetchExchange, Provider } from 'urql';
 
+import { supabaseClient } from 'src/context/GlobalProviders';
 import { useUserStore } from 'src/context/User/useUserContextStore';
 import useDataFetchErrorHandling from 'src/hooks/useDataFetchErrorHandling';
+import { logRocketEvent } from 'src/services/shared';
 import { getAuthHeader } from 'src/utils/misc-utils';
 
 function UrqlConfigProvider({ children }: BaseComponentProps) {
     const { checkIfAuthInvalid, forceUserToSignOut } =
         useDataFetchErrorHandling();
 
-    const [accessToken, expiresAt] = useUserStore((state) => [
-        state.session?.access_token,
-        state.session?.expires_at,
-    ]);
+    const [accessToken, expiresAt, refreshToken] = useUserStore(
+        useShallow((state) => [
+            state.session?.access_token,
+            state.session?.expires_at,
+            state.session?.refresh_token,
+        ])
+    );
```

Comment:
Adding `refresh token` but also using `shallow` now as these are all strings and Supabase "refreshes" the session object every time the browser regains focus.